client/state/media/reducer.js

```diff
@@ -74,7 +78,41 @@ export const queryRequests = createReducer( {}, {
 	}
 } );
 
+/**
+ * Returns the updated site post requests state after an action has been
+ * dispatched. The state reflects a mapping of site ID, post ID pairing to a
+ * boolean reflecting whether a request for the post is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const mediaItemRequests = createReducer( {}, {
+	[ MEDIA_ITEM_REQUESTING ]: ( state, { siteId, mediaId } ) => {
+		return {
+			...state,
+			[ siteId ]: {
+				...state[ siteId ],
+				[ mediaId ]: true
+			}
+		};
+	},
+	[ MEDIA_ITEM_REQUEST_SUCCESS ]: ( state, { siteId, mediaId } ) => {
```

Comment:
In above `queryRequests`, we flag request as having completed when receiving media. We should probably have a separate action `MEDIA_REQUEST_SUCCESS`, which itself would be fine to defer to a future pull request. What I might wonder is since we also trigger `MEDIA_RECEIVE` when requesting a single item, does it do anything funky with how it tries to omit from state using `MediaQueryManager.QueryKey.stringify` of an `undefined` query?